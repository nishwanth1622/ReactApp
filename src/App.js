import React, { BrowserRouter as Router, Route, Routes, Link, useParams } from 'react-router-dom';
import './App.css';
import HomePage from './HomePage';
import AboutUs from './AboutUs';
import Contact from './Contact';
import Services from './Services';
import Register from './Register';
import PersonComponent from './PersonComponent';
//import Employee from './Employee';
import NameComponent from './NameComponent';
import Search from'./Search';


function App() { //Es5

  const Wrapper = (props) => {
    const params = useParams();
    return <Services {...{ ...props, match: { params } }} />
  }

  return (
    <div className="App">

      <Router>
        <ul>
          <button type="button">
            <li>
              <Link to="/">HomePage</Link>
            </li>
          </button>
          <button type="button">
            <li>
              <Link to="/AboutUs">AboutUs</Link>
            </li>
          </button>
          <button type="button">
            <li>
              <Link to="/Services/Main">Services</Link>
            </li>
          </button>
          <button type="button">
            <li>
              <Link to="/Contact">Contact</Link>
            </li>
          </button>
          <button type="button">
            <li>
              <Link to="/Register">Register</Link>
            </li>
          </button>
          {/* <button type="button">
            <li>
              <Link to="/employees">Employees</Link>
            </li>
          </button> */}
          <button type="search">
            <li>
              <Link to='/search'>Search</Link>
            </li>
          </button>

        </ul>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/AboutUs" element={<AboutUs />} />
          {/* <Route path="/employees" element={<Employee />} /> */}
          <Route path="/Services/:id" element={<Wrapper />} />
          <Route path="/Contact" element={<Contact />} />
          <Route path="/Register" element={<Register />} />
          <Route path="/persons" element={<PersonComponent />} />
          <Route path="/customers" element={<NameComponent />} />
          <Route path="/Search" element={<Search />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
