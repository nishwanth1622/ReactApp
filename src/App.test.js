import { render, screen } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';

test('renders learn react link', () => {
  // render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
});

test('renders basic Home page', () => {

  const wrapper = render(<App />)
  expect(wrapper).toMatchSnapshot()
});
