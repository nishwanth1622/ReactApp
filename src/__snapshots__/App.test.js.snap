// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <ul>
          <button
            type="button"
          >
            <li>
              <a
                href="/"
              >
                HomePage
              </a>
            </li>
          </button>
          <button
            type="button"
          >
            <li>
              <a
                href="/AboutUs"
              >
                AboutUs
              </a>
            </li>
          </button>
          <button
            type="button"
          >
            <li>
              <a
                href="/Services/Main"
              >
                Services
              </a>
            </li>
          </button>
          <button
            type="button"
          >
            <li>
              <a
                href="/Contact"
              >
                Contact
              </a>
            </li>
          </button>
          <button
            type="button"
          >
            <li>
              <a
                href="/Register"
              >
                Register
              </a>
            </li>
          </button>
        </ul>
        <h1>
          Indusvalley INC
        </h1>
        <img
          alt="react logo"
          src="Indusvalley.jpeg"
        />
        <h4>
           Trust comes with time
        </h4>
        <p>
          Everyday at Indusvalley, we wake up with one sole focus - to provide our clients with the best IT solutions and help them optimize their operations. 
          <br />
          Since 1996, we have devoted our expertise to understanding client needs to find the best solution or build them one if none is available. We simply don't believe in impossible challenges. We are especially adept at offering our clients customized solutions for their unique needs. And since trust comes with time, Indus Valley has grown to become the benchmark for quality IT consulting over the 25 years we have been in existence. No matter what you need, you can always rely on our experts to deliver over and above your expectations.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <ul>
        <button
          type="button"
        >
          <li>
            <a
              href="/"
            >
              HomePage
            </a>
          </li>
        </button>
        <button
          type="button"
        >
          <li>
            <a
              href="/AboutUs"
            >
              AboutUs
            </a>
          </li>
        </button>
        <button
          type="button"
        >
          <li>
            <a
              href="/Services/Main"
            >
              Services
            </a>
          </li>
        </button>
        <button
          type="button"
        >
          <li>
            <a
              href="/Contact"
            >
              Contact
            </a>
          </li>
        </button>
        <button
          type="button"
        >
          <li>
            <a
              href="/Register"
            >
              Register
            </a>
          </li>
        </button>
      </ul>
      <h1>
        Indusvalley INC
      </h1>
      <img
        alt="react logo"
        src="Indusvalley.jpeg"
      />
      <h4>
         Trust comes with time
      </h4>
      <p>
        Everyday at Indusvalley, we wake up with one sole focus - to provide our clients with the best IT solutions and help them optimize their operations. 
        <br />
        Since 1996, we have devoted our expertise to understanding client needs to find the best solution or build them one if none is available. We simply don't believe in impossible challenges. We are especially adept at offering our clients customized solutions for their unique needs. And since trust comes with time, Indus Valley has grown to become the benchmark for quality IT consulting over the 25 years we have been in existence. No matter what you need, you can always rely on our experts to deliver over and above your expectations.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders learn react link 1`] = `ShallowWrapper {}`;
